src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -656,56 +672,60 @@ where
         Ok(version)
     }
 
-    // TODO(Yuanxin): Use this function for both `CREATE TABLE` and `CREATE TABLE WITH CONNECTOR`.
-    async fn create_materialized_source_inner(
+    /// If `source` is `None` here, then there is no external streaming source for this table.
+    async fn create_table_inner(
         &self,
-        mut source: Source,
+        mut source: Option<Source>,
         mut mview: Table,
         mut fragment_graph: StreamFragmentGraph,
-    ) -> MetaResult<(SourceId, TableId, CatalogVersion)> {
+    ) -> MetaResult<(TableId, CatalogVersion)> {
         self.check_barrier_manager_status().await?;
 
-        // Generate source id.
-        let source_id = self.gen_unique_id::<{ IdCategory::Table }>().await?; // TODO: use source category
-        source.id = source_id;
+        if let Some(source) = &mut source {
+            // Generate source id.
+            let source_id = self.gen_unique_id::<{ IdCategory::Table }>().await?; // TODO: Use source category
+            source.id = source_id;
```

Comment:
IIUC, if you mean something like: `create table as select * from source`, this should be done by `create materialized view as select * from source`, and this will go to `create_materialized_view` instead of here (`create_table`).

cc @st1page

Reply:
I mean, is it possible to `CREATE SOURCE source WITH (..)` and then create a DML-able table on top of this `source`, without specifying the properties again? 🤔

Reply:
> I mean, is it possible to `CREATE SOURCE source WITH (..)` and then create a DML-able table on top of this `source`, without specifying the properties again? 🤔

I think it's possible and we should support it from the user's perspective. In this case we don't have to fill the source id if its relied source already created.